### PR TITLE
Faster tests / Github CI / Fixes

### PR DESCRIPTION
**Pull request by gpotter2**
- move non root tests to GithubCI: they take ~ 8 minutes in total. Speeds up the travis builds by a lot & give instant feedback to PRs
- fix tests failures after guedou's tests: https://github.com/secdev/scapy/pull/2541 (tox would fail silently)
- pin sphinx >= 2.4.2 (contains fix to https://github.com/secdev/scapy/issues/2475)
- Update Github workflow for better names
- add non root keyword to UTscapy: `-N`
- work around https://github.com/secdev/scapy/issues/2542 (still needs to be fixed !)

### CODE REVIEW

File: .config/ci/install.sh [hunk 1]
```diff
@@ -29,8 +29,6 @@ python -m pip install --upgrade pip setuptools --ignore-installed
 # Make sure tox is installed and up to date
 python -m pip install -U tox --ignore-installed
 
-# Make sure brotli is installed and up to date
-python -m pip install -U brotli --ignore-installed
```
Comment (guedou) [inline .config/ci/install.sh]: Why are you removing this?

File: .config/ci/test.sh [hunk 2]
```diff
@@ -67,13 +68,13 @@ echo UT_FLAGS=$UT_FLAGS
 echo TOXENV=$TOXENV
 
 # Launch Scapy unit tests
-tox -- ${UT_FLAGS}
+tox -- ${UT_FLAGS} || exit 1;
```
Comment (guedou) [inline .config/ci/test.sh]: s/;//

File: .config/ci/test.sh [hunk 3]
```diff
@@ -67,13 +68,13 @@ echo UT_FLAGS=$UT_FLAGS
...
 
 # Start Scapy in interactive mode
 TEMPFILE=$(mktemp)
-cat << EOF > ${TEMPFILE}
+cat << EOF > "${TEMPFILE}"
 print("Scapy on %s" % sys.version)
 sys.exit()
 EOF
-./run_scapy -H -c ${TEMPFILE}
+./run_scapy -H -c "${TEMPFILE}" || exit 1;
```
Comment (guedou) [inline .config/ci/test.sh]: s/;//

File: .config/ci/install.sh (hunk 1 above)
Comment (gpotter2) [reply .config/ci/install.sh]: It's already handled by tox

**Pull request merged**